selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/preferences/ContributeTabTest.java

```diff
@@ -0,0 +1,288 @@
+
+    // check the 'Refresh button
+    preferences.clickOnContributeCheckbox();
+    preferences.waitContributeCheckboxIsNotSelected();
+
+    assertTrue(preferences.saveButtonIsEnabled());
+
+    preferences.clickRefreshButton();
+    preferences.waitContributeCheckboxIsSelected();
+
+    assertFalse(
+        preferences.saveButtonIsEnabled(),
+        "Known issue https://github.com/eclipse/che/issues/9959");
+
+    preferences.clickOnCloseBtnAndWaitClosedForm();
+  }
+
+  @Test(priority = 1)
+  public void checkSwitchProjectsWhenContributeIsFalse() {
+    openContributeTab();
+    setStateContributeChecboxAndCloseForm(true);
+
+    projectExplorer.waitAndSelectItem(FIRST_PROJECT_NAME);
+    pullRequestPanel.waitOpenPanel();
+
+    openContributeTab();
+    setStateContributeChecboxAndCloseForm(false);
+
+    // switch between projects
+    projectExplorer.waitAndSelectItem(FIRST_PROJECT_NAME);
+    pullRequestPanel.waitRepoUrl(firstProjectUrl);
+    pullRequestPanel.waitProjectName(FIRST_PROJECT_NAME);
+
+    projectExplorer.waitAndSelectItem(SECOND_PROJECT_NAME);
+    pullRequestPanel.waitRepoUrl(secondProjectUrl);
+    pullRequestPanel.waitProjectName(SECOND_PROJECT_NAME);
+
+    projectExplorer.waitAndSelectItem(THIRD_PROJECT_NAME);
+    pullRequestPanel.waitTetxNotVcsProject(EXP_TEXT_NOT_VCS);
+  }
+
+  @Test(priority = 1)
+  public void checkAutomaticChangeContributeToFalse() {
+    openContributeTab();
+    setStateContributeChecboxAndCloseForm(true);
+
+    // change 'Contribute' to false by 'Hide' button
+    projectExplorer.waitAndSelectItem(FIRST_PROJECT_NAME);
+    pullRequestPanel.waitOpenPanel();
+    pullRequestPanel.clickHideButtonAndWaitClosePanel();
+    notificationsPopupPanel.waitExpectedMessageOnProgressPanelAndClosed(NOTIFICATION_MESS);
+
+    openContributeTab();
+    preferences.waitContributeCheckboxIsNotSelected();
+
+    // change 'Contribute' to false by 'Hide' from 'Options' on the PR panel
+    setStateContributeChecboxAndCloseForm(true);
+
+    projectExplorer.waitAndSelectItem(SECOND_PROJECT_NAME);
+    pullRequestPanel.waitOpenPanel();
+    pullRequestPanel.openOptionsMenu();
+    pullRequestPanel.clickContextHideButtonAndWAitClosedPanel();
+    notificationsPopupPanel.waitExpectedMessageOnProgressPanelAndClosed(NOTIFICATION_MESS);
+
+    openContributeTab();
+    preferences.waitContributeCheckboxIsNotSelected();
+
+    // change 'Contribute' to false by 'Contribute' action from 'Assistant'
+    setStateContributeChecboxAndCloseForm(true);
+
+    projectExplorer.waitAndSelectItem(FIRST_PROJECT_NAME);
+    pullRequestPanel.waitOpenPanel();
+    pullRequestPanel.clickPullRequestBtn();
+
+    menu.runCommand(ASSISTANT, TOOL_WINDOWS, CONTRIBUTE_TOOL_WIDOWS);
+    pullRequestPanel.waitClosePanel();
+
+    openContributeTab();
+    preferences.waitContributeCheckboxIsNotSelected();
+    preferences.clickOnCloseBtnAndWaitClosedForm();
+  }
+
+  @Test(priority = 1)
+  public void checkOpenPrPanelOnDemand() {
```

Comment:
`checkOpenPrPanelOnDemand` test name is not clear as for me

Reply:
When the property "Activate Contribute Panel by clicking on a project" is false (disabled), an user can display the panel:
- change the preference property to "true" and select a project
- using Panel selector
- from menu Assistant ---> Tool Windows ---> Contribute action
- by hot key

PR #9806

Reply:
In that case the name could be rewritten in more clear view: `checkManualOperationsWithContributeTab`.

Reply:
I agree